### ISSUE TITLE
IDE0009 reported on collection initializers with unqualifiable items

### ISSUE THREAD

**Issue by adimosh** [dotnet/roslyn#30206]
**Version Used**: 
Visual Studio 2017 15.8
Visual Studio 2017 15.7

**Steps to Reproduce**:

1. Declare a collection and use the collection initializer syntax
2. Ensure that the collection is populated with unqualifiable items, for example, a collection of types initialized with multiple lines of typeof(SomeType), or a collection of constant values.

This works for both simple collection initializers, as well as the standard Dictionary<TKey, TValue> and its variations, and also custom collections or dictionaries with Add methods having multiple parameters. For example:

```
var unaryOperators = new LevelDictionary<string, Type>
{
    { this.Definition.SubtractSymbol, typeof(Nodes.Operations.Unary.SubtractNode), 1 },
    { this.Definition.NotSymbol, typeof(Nodes.Operations.Unary.NotNode), 1 },
};
```

**Expected Behavior**:
The set of lines is considered correct.

**Actual Behavior**:
The lines will be squiggled, and "Add this. or Me." will be given as a suggestion.

If one attempts to do the analyzer's code correction, nothing will happen in the file, however some sort of undoable action is performed, as undo or redo treat the action as if something has happened.

**Comment by hikeeba** [comment]
I am seeing this actually add the `this.` prefix when the whole statement is on one line and it breaks the build.

**Comment by jinujoseph** [comment]
This looks same as the fix we took in 15.9 [link](https://github.com/dotnet/roslyn/pull/29449)
Let us know if you think otherwise
